ci/benchmark_parser.py

```diff
@@ -0,0 +1,155 @@
+
+    :param directory: path to directory that contains raw results as :class:`pathlib.Path`
+    :param name_suffix: a :class:`str` suffix to apply to each test name found
+
+    :return: :class:`list` of data points
+    """
+    excluded_directories = ["child_generate", "fork", "parent_generate", "report"]
+    result_values = list()
+    for dire in directory.iterdir():
+        if dire.name in excluded_directories or not dire.is_dir():
+            continue
+        for subdir in dire.iterdir():
+            if subdir.name != "new":
+                continue
+
+            test_name = parse_benchmark_file(subdir)
+            for stat_name, value in parse_estimate_file(subdir).items():
+                test_name_parts = list(filter(None, [test_name, stat_name, name_suffix]))
+                result_values.append({"value": value, "test": "_".join(test_name_parts)})
+
+    return result_values
+
+
+def parse_benchmark_file(directory):
+    """
+    Parse file containing details of the parameters used for a benchmark.
+
+    :param directory: directory where a benchmark case results are located as :class:`pathlib.Path`
+
+    :return: name of the test as :class:`str`
+    """
+    raw_results = _parse_file_to_json(directory, "benchmark.json")
+    return raw_results["full_id"].replace(" ", "_")
+
+
+def parse_estimate_file(directory):
+    """
+    Parse file containing timing results for a benchmark.
+
+    :param directory: directory where a benchmark case results are located as :class:`pathlib.Path`
+
+    :return: :class:`dict` of data points
+    """
+    raw_results = _parse_file_to_json(directory, "estimates.json")
+    return {
+        stat_name: raw_results[stat_name]["point_estimate"]
+        for stat_name in ("mean", "std_dev")
+    }
+
+
+def _parse_file_to_json(directory, filename):
+    result_file = directory.joinpath(filename)
+    return json.loads(result_file.read_text())
+
+
+def dump_results(parsed_results, filename, input_args):
+    """
+    Dump parsed results formatted as JSON to file.
+
+    :param parsed_results: :class:`list` of data points
+    :param filename: filename for dump file as :class:`pathlib.Path`
+    :param input_args: CLI input arguments
+    """
+    if input_args.append_results:
+        parsed_content = json.loads(filename.read_text())
+        parsed_content["points"].extend(parsed_results)
+        filename.write_text(json.dumps(parsed_content))
+    else:
+        filename.parent.mkdir(parents=True, exist_ok=True)
+        series = {
+            "schema": input_args.schema,
+            "hardware": input_args.hardware,
+            "project_version": input_args.project_version,
+            "branch": input_args.branch,
+            "insert_date": input_args.bench_date,
+            "commit_date": input_args.commit_date,
+            "points": parsed_results,
+        }
+        filename.write_text(json.dumps(series))
+
+
+def check_mandatory_args(input_args):
```

Comment:
you can add a keyword required=True to the argparse to make arguments mandatory

Reply:
Not really since I want to ignore those arguments if `--append-results` flag is set. I've thought about using a mutual exclusive group in  `argparse` but that won't to the job for my need.

Reply:
yep understood, looks good then